File: wardrobe_organizer.go
# wardrobe_organizer.go
/**
 * 👗 Wardrobe Organizer – Smart Clothing Manager (Go Edition)
 * Features: add items, categories, color stats, outfit builder, random outfit
 */

package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"math/rand"
	"os"
	"path/filepath"
	"strings"
	"time"
	"github.com/google/uuid"
)

// ─── Types ──────────────────────────────────────────────────────────────────

type Item struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	Category string `json:"category"`
	Color    string `json:"color"`
	Season   string `json:"season"`
	Photo    string `json:"photo"`
	Created  string `json:"created"`
}

type Outfit struct {
	ID      string   `json:"id"`
	Name    string   `json:"name"`
	Items   []string `json:"items"`
	Created string   `json:"created"`
}

type Data struct {
	Items   []Item   `json:"items"`
	Outfits []Outfit `json:"outfits"`
}

// ─── Colors ──────────────────────────────────────────────────────────────────

const (
	reset  = "\x1b[0m"
	bright = "\x1b[1m"
	dim    = "\x1b[2m"
	red    = "\x1b[31m"
	green  = "\x1b[32m"
	yellow = "\x1b[33m"
	blue   = "\x1b[34m"
	magenta = "\x1b[35m"
	cyan   = "\x1b[36m"
)

func c(str, color string) string {
	return color + str + reset
}

// ─── Constants ──────────────────────────────────────────────────────────────

var CATEGORIES = []string{"Tops", "Bottoms", "Shoes", "Accessories", "Outerwear", "Dresses"}
var COLORS = []string{"Black", "White", "Red", "Blue", "Green", "Yellow", "Purple", "Pink",
	"Orange", "Brown", "Grey", "Navy", "Beige", "Olive", "Burgundy"}
var SEASONS = []string{"Spring", "Summer", "Autumn", "Winter", "All"}
var COLOR_EMOJIS = map[string]string{
	"Black": "⚫", "White": "⚪", "Red": "🔴", "Blue": "🔵", "Green": "🟢",
	"Yellow": "🟡", "Purple": "🟣", "Pink": "💗", "Orange": "🟠",
	"Brown": "🟤", "Grey": "◻️", "Navy": "💙", "Beige": "🟫", "Olive": "🫒",
}

// ─── Data Manager ──────────────────────────────────────────────────────────

type WardrobeData struct {
	filePath string
	Items    []Item
	Outfits  []Outfit
}

func NewWardrobeData() *WardrobeData {
	home, _ := os.UserHomeDir()
	dir := filepath.Join(home, ".wardrobe")
	os.MkdirAll(dir, 0755)
	filePath := filepath.Join(dir, "data.json")
	wd := &WardrobeData{filePath: filePath}
	wd.load()
	return wd
}

func (wd *WardrobeData) load() {
	if _, err := os.Stat(wd.filePath); os.IsNotExist(err) {
		return
	}
	raw, err := os.ReadFile(wd.filePath)
	if err != nil {
		return
	}
	var data Data
	if err := json.Unmarshal(raw, &data); err != nil {
		return
	}
	wd.Items = data.Items
	wd.Outfits = data.Outfits
}

func (wd *WardrobeData) save() {
	data := Data{Items: wd.Items, Outfits: wd.Outfits}
	raw, _ := json.MarshalIndent(data, "", "  ")
	os.WriteFile(wd.filePath, raw, 0644)
}

func (wd *WardrobeData) AddItem(name, category, color, season, photo string) Item {
	item := Item{
		ID:       uuid.New().String(),
		Name:     name,
		Category: category,
		Color:    color,
		Season:   season,
		Photo:    photo,
		Created:  time.Now().Format(time.RFC3339),
	}
	wd.Items = append(wd.Items, item)
	wd.save()
	return item
}

func (wd *WardrobeData) DeleteItem(id string) bool {
	for i, item := range wd.Items {
		if item.ID == id {
			// Remove from outfits
			for j := range wd.Outfits {
				for k, itemID := range wd.Outfits[j].Items {
					if itemID == id {
						wd.Outfits[j].Items = append(wd.Outfits[j].Items[:k], wd.Outfits[j].Items[k+1:]...)
						break
					}
				}
			}
			wd.Items = append(wd.Items[:i], wd.Items[i+1:]...)
			wd.save()
			return true
		}
	}
	return false
}

func (wd *WardrobeData) GetItem(id string) *Item {
	for _, item := range wd.Items {
		if item.ID == id {
			return &item
		}
	}
	return nil
}

func (wd *WardrobeData) AddOutfit(name string, itemIDs []string) Outfit {
	outfit := Outfit{
		ID:      uuid.New().String(),
		Name:    name,
		Items:   itemIDs,
		Created: time.Now().Format(time.RFC3339),
	}
	wd.Outfits = append(wd.Outfits, outfit)
	wd.save()
	return outfit
}

func (wd *WardrobeData) SearchItems(query, category, color, season string) []Item {
	results := wd.Items
	if query != "" {
		q := strings.ToLower(query)
		results = filterItems(results, func(i Item) bool {
			return strings.Contains(strings.ToLower(i.Name), q) ||
				strings.Contains(strings.ToLower(i.Category), q)
		})
	}
	if category != "" {
		c := strings.ToLower(category)
		results = filterItems(results, func(i Item) bool {
			return strings.ToLower(i.Category) == c
		})
	}
	if color != "" {
		c := strings.ToLower(color)
		results = filterItems(results, func(i Item) bool {
			return strings.ToLower(i.Color) == c
		})
	}
	if season != "" {
		s := strings.ToLower(season)
		results = filterItems(results, func(i Item) bool {
			return strings.ToLower(i.Season) == s || strings.ToLower(i.Season) == "all"
		})
	}
	return results
}

func filterItems(items []Item, fn func(Item) bool) []Item {
	var res []Item
	for _, i := range items {
		if fn(i) {
			res = append(res, i)
		}
	}
	return res
}

func (wd *WardrobeData) GetColorStats() map[string]int {
	stats := make(map[string]int)
	for _, i := range wd.Items {
		stats[i.Color]++
	}
	return stats
}

func (wd *WardrobeData) GetCategoryStats() map[string]int {
	stats := make(map[string]int)
	for _, i := range wd.Items {
		stats[i.Category]++
	}
	return stats
}

func (wd *WardrobeData) GetSeasonStats() map[string]int {
	stats := make(map[string]int)
	for _, i := range wd.Items {
		stats[i.Season]++
	}
	return stats
}

func (wd *WardrobeData) RandomOutfit(season string) []Item {
	items := wd.Items
	if season != "" && strings.ToLower(season) != "all" {
		s := strings.ToLower(season)
		items = filterItems(items, func(i Item) bool {
			return strings.ToLower(i.Season) == s || strings.ToLower(i.Season) == "all"
		})
	}
	if len(items) == 0 {
		return []Item{}
	}
	var outfit []Item
	for _, cat := range CATEGORIES {
		var catItems []Item
		for _, i := range items {
			if i.Category == cat {
				catItems = append(catItems, i)
			}
		}
		if len(catItems) > 0 {
			outfit = append(outfit, catItems[rand.Intn(len(catItems))])
		}
	}
	return outfit
}

// ─── Main App ──────────────────────────────────────────────────────────────

type WardrobeApp struct {
	reader *bufio.Reader
	data   *WardrobeData
}

func NewWardrobeApp() *WardrobeApp {
	rand.Seed(time.Now().UnixNano())
	return &WardrobeApp{
		reader: bufio.NewReader(os.Stdin),
		data:   NewWardrobeData(),
	}
}

func (app *WardrobeApp) ask(prompt string) string {
	fmt.Print(prompt)
	line, _ := app.reader.ReadString('\n')
	return strings.TrimSpace(line)
}

func (app *WardrobeApp) showMenu() {
	total := len(app.data.Items)
	categories := len(app.data.GetCategoryStats())
	fmt.Println("\n" + c(strings.Repeat("═", 50), cyan))
	fmt.Println(c("👗 WARDROBE ORGANIZER", bright+cyan))
	fmt.Println(c(strings.Repeat("═", 50), cyan))
	fmt.Printf("  Items: %d\n", total)
	fmt.Printf("  Categories: %d\n", categories)
	fmt.Printf("  Outfits: %d\n", len(app.data.Outfits))
	fmt.Println(c(strings.Repeat("═", 50), cyan))
	fmt.Println("  1. 👕 Add clothing item")
	fmt.Println("  2. 📋 List all items")
	fmt.Println("  3. 🔍 Search items")
	fmt.Println("  4. 📊 Statistics")
	fmt.Println("  5. 🎭 Outfit Builder")
	fmt.Println("  6. 🎲 Random Outfit")
	fmt.Println("  7. ❤️  Favorites (coming soon)")
	fmt.Println("  8. 🗑️  Delete item")
	fmt.Println("  0. 🚪 Exit")
	fmt.Println(c(strings.Repeat("═", 50), cyan))
}

func (app *WardrobeApp) listItems(items []Item) {
	if items == nil {
		items = app.data.Items
	}
	if len(items) == 0 {
		fmt.Println(c("No items found.", yellow))
		return
	}
	fmt.Println("\n👕 WARDROBE ITEMS")
	fmt.Println(c(strings.Repeat("─", 60), dim))
	for i, item := range items {
		emoji := COLOR_EMOJIS[item.Color]
		fmt.Printf("  %d. %s %s (%s) %s %s\n", i+1, emoji, item.Name, item.Category, item.Color, item.Season)
	}
}

func (app *WardrobeApp) addItem() {
	name := app.ask("Item name: ")
	fmt.Printf("Categories: %s\n", strings.Join(CATEGORIES, ", "))
	category := app.ask("Category: ")
	fmt.Printf("Colors: %s\n", strings.Join(COLORS, ", "))
	color := app.ask("Color: ")
	fmt.Printf("Seasons: %s\n", strings.Join(SEASONS, ", "))
	season := app.ask("Season: ")
	photo := app.ask("Photo path (optional): ")
	item := app.data.AddItem(name, category, color, season, photo)
	emoji := COLOR_EMOJIS[color]
	fmt.Printf("%s\n", c(fmt.Sprintf("✅ Added %s %s (%s, %s)", emoji, item.Name, item.Category, item.Color), green))
}

func (app *WardrobeApp) searchItems() {
	query := app.ask("Search term (name/category): ")
	category := app.ask("Filter by category (optional): ")
	color := app.ask("Filter by color (optional): ")
	season := app.ask("Filter by season (optional): ")
	results := app.data.SearchItems(query, category, color, season)
	if len(results) > 0 {
		app.listItems(results)
	} else {
		fmt.Println(c("No items match your search.", yellow))
	}
}

func (app *WardrobeApp) showStats() {
	colorStats := app.data.GetColorStats()
	categoryStats := app.data.GetCategoryStats()
	seasonStats := app.data.GetSeasonStats()
	fmt.Println("\n📊 STATISTICS")
	fmt.Println(c(strings.Repeat("─", 30), dim))
	fmt.Println("\n🎨 Colors:")
	for color, count := range colorStats {
		emoji := COLOR_EMOJIS[color]
		fmt.Printf("  %s %s: %d\n", emoji, color, count)
	}
	fmt.Println("\n📂 Categories:")
	for cat, count := range categoryStats {
		fmt.Printf("  %s: %d\n", cat, count)
	}
	fmt.Println("\n🌦️ Seasons:")
	for season, count := range seasonStats {
		fmt.Printf("  %s: %d\n", season, count)
	}
}

func (app *WardrobeApp) outfitBuilder() {
	if len(app.data.Items) == 0 {
		fmt.Println(c("No items. Add some clothing first!", yellow))
		return
	}
	name := app.ask("Outfit name: ")
	fmt.Println("Select items for your outfit:")
	app.listItems(nil)
	var itemIDs []string
	for {
		choice := app.ask("Enter item number to add (or 'done'): ")
		if strings.ToLower(choice) == "done" {
			break
		}
		var idx int
		fmt.Sscanf(choice, "%d", &idx)
		idx--
		if idx >= 0 && idx < len(app.data.Items) {
			item := app.data.Items[idx]
			found := false
			for _, id := range itemIDs {
				if id == item.ID {
					found = true
					break
				}
			}
			if !found {
				itemIDs = append(itemIDs, item.ID)
				fmt.Printf("%s\n", c("Added "+item.Name, green))
			} else {
				fmt.Println(c("Already added", yellow))
			}
		} else {
			fmt.Println(c("Invalid number", red))
		}
	}
	if len(itemIDs) > 0 {
		outfit := app.data.AddOutfit(name, itemIDs)
		fmt.Printf("%s\n", c(fmt.Sprintf("✅ Outfit '%s' created with %d items", outfit.Name, len(itemIDs)), green))
		app.showOutfit(outfit)
	}
}

func (app *WardrobeApp) showOutfit(outfit Outfit) {
	fmt.Printf("\n🎭 %s\n", outfit.Name)
	for _, id := range outfit.Items {
		item := app.data.GetItem(id)
		if item != nil {
			emoji := COLOR_EMOJIS[item.Color]
			fmt.Printf("  %s %s (%s) %s\n", emoji, item.Name, item.Category, item.Color)
		}
	}
}

func (app *WardrobeApp) randomOutfit() {
	if len(app.data.Items) == 0 {
		fmt.Println(c("No items. Add some clothing first!", yellow))
		return
	}
	season := app.ask("Season (optional, press Enter for all): ")
	outfit := app.data.RandomOutfit(season)
	if len(outfit) == 0 {
		fmt.Println(c("No items for this season.", yellow))
		return
	}
	fmt.Println("\n🎲 Random Outfit")
	for _, item := range outfit {
		emoji := COLOR_EMOJIS[item.Color]
		fmt.Printf("  %s %s (%s) %s\n", emoji, item.Name, item.Category, item.Color)
	}
}

func (app *WardrobeApp) deleteItem() {
	if len(app.data.Items) == 0 {
		fmt.Println(c("No items to delete.", yellow))
		return
	}
	app.listItems(nil)
	choice := app.ask("Enter item number to delete (or 'cancel'): ")
	if strings.ToLower(choice) == "cancel" {
		return
	}
	var idx int
	fmt.Sscanf(choice, "%d", &idx)
	idx--
	if idx >= 0 && idx < len(app.data.Items) {
		item := app.data.Items[idx]
		confirm := app.ask(fmt.Sprintf("Delete '%s'? (yes/no): ", item.Name))
		if strings.ToLower(confirm) == "yes" {
			app.data.DeleteItem(item.ID)
			fmt.Printf("%s\n", c("🗑️  Deleted "+item.Name, yellow))
		}
	} else {
		fmt.Println(c("Invalid number", red))
	}
}

func (app *WardrobeApp) run() {
	fmt.Print("\033[H\033[2J")
	fmt.Printf("%s\n", c("\n👗 Wardrobe Organizer – Smart Clothing Manager", bright+cyan))
	fmt.Printf("%s\n", c("Catalog your wardrobe, plan outfits!", dim))

	for {
		app.showMenu()
		choice := app.ask("Your choice: ")
		switch choice {
		case "1":
			app.addItem()
		case "2":
			app.listItems(nil)
		case "3":
			app.searchItems()
		case "4":
			app.showStats()
		case "5":
			app.outfitBuilder()
		case "6":
			app.randomOutfit()
		case "7":
			fmt.Println(c("❤️ Favorites feature coming soon!", dim))
		case "8":
			app.deleteItem()
		case "0":
			fmt.Printf("%s\n", c("👋 Stay stylish! Goodbye!", cyan))
			return
		default:
			fmt.Println(c("❌ Invalid choice.", red))
		}
		if choice != "0" {
			fmt.Print("\nPress Enter to continue...")
			app.reader.ReadString('\n')
		}
	}
}

func main() {
	app := NewWardrobeApp()
	app.run()
}

File: wardrobe_organizer.rs
# wardrobe_organizer.rs
/**
 * 👗 Wardrobe Organizer – Smart Clothing Manager (Rust Edition)
 * Features: add items, categories, color stats, outfit builder, random outfit
 * Dependencies: serde, serde_json, chrono, colored, rand, uuid
 */

use chrono::Utc;
use colored::*;
use rand::seq::SliceRandom;
use rand::thread_rng;
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use std::fs;
use std::io::{self, Write, BufRead};
use std::path::PathBuf;
use uuid::Uuid;

// ─── Types ──────────────────────────────────────────────────────────────────

#[derive(Debug, Serialize, Deserialize, Clone)]
struct Item {
    id: String,
    name: String,
    category: String,
    color: String,
    season: String,
    photo: String,
    created: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
struct Outfit {
    id: String,
    name: String,
    items: Vec<String>,
    created: String,
}

#[derive(Debug, Serialize, Deserialize)]
struct Data {
    items: Vec<Item>,
    outfits: Vec<Outfit>,
}

// ─── Colors ──────────────────────────────────────────────────────────────────

fn c(text: &str, color: &str) -> String {
    match color {
        "green" => text.green().to_string(),
        "red" => text.red().to_string(),
        "yellow" => text.yellow().to_string(),
        "cyan" => text.cyan().to_string(),
        "bright" => text.bright().to_string(),
        "dim" => text.dimmed().to_string(),
        _ => text.to_string(),
    }
}

// ─── Constants ──────────────────────────────────────────────────────────────

const CATEGORIES: &[&str] = &["Tops", "Bottoms", "Shoes", "Accessories", "Outerwear", "Dresses"];
const COLORS: &[&str] = &["Black", "White", "Red", "Blue", "Green", "Yellow", "Purple", "Pink",
                         "Orange", "Brown", "Grey", "Navy", "Beige", "Olive", "Burgundy"];
const SEASONS: &[&str] = &["Spring", "Summer", "Autumn", "Winter", "All"];

lazy_static! {
    static ref COLOR_EMOJIS: HashMap<String, String> = {
        let mut m = HashMap::new();
        m.insert("Black".to_string(), "⚫".to_string());
        m.insert("White".to_string(), "⚪".to_string());
        m.insert("Red".to_string(), "🔴".to_string());
        m.insert("Blue".to_string(), "🔵".to_string());
        m.insert("Green".to_string(), "🟢".to_string());
        m.insert("Yellow".to_string(), "🟡".to_string());
        m.insert("Purple".to_string(), "🟣".to_string());
        m.insert("Pink".to_string(), "💗".to_string());
        m.insert("Orange".to_string(), "🟠".to_string());
        m.insert("Brown".to_string(), "🟤".to_string());
        m.insert("Grey".to_string(), "◻️".to_string());
        m.insert("Navy".to_string(), "💙".to_string());
        m.insert("Beige".to_string(), "🟫".to_string());
        m.insert("Olive".to_string(), "🫒".to_string());
        m
    };
}

// ─── Data Manager ──────────────────────────────────────────────────────────

struct WardrobeData {
    file_path: PathBuf,
    items: Vec<Item>,
    outfits: Vec<Outfit>,
}

impl WardrobeData {
    fn new() -> Self {
        let home = std::env::var("HOME").or_else(|_| std::env::var("USERPROFILE")).unwrap_or_else(|_| ".".to_string());
        let dir = PathBuf::from(home).join(".wardrobe");
        fs::create_dir_all(&dir).unwrap();
        let file_path = dir.join("data.json");
        let mut wd = WardrobeData { file_path, items: Vec::new(), outfits: Vec::new() };
        wd.load();
        wd
    }

    fn load(&mut self) {
        if let Ok(raw) = fs::read_to_string(&self.file_path) {
            if let Ok(data) = serde_json::from_str::<Data>(&raw) {
                self.items = data.items;
                self.outfits = data.outfits;
                return;
            }
        }
    }

    fn save(&self) {
        let data = Data { items: self.items.clone(), outfits: self.outfits.clone() };
        let raw = serde_json::to_string_pretty(&data).unwrap();
        let _ = fs::write(&self.file_path, raw);
    }

    fn add_item(&mut self, name: String, category: String, color: String, season: String, photo: String) -> Item {
        let item = Item {
            id: Uuid::new_v4().to_string(),
            name,
            category,
            color,
            season,
            photo,
            created: Utc::now().to_rfc3339(),
        };
        self.items.push(item.clone());
        self.save();
        item
    }

    fn delete_item(&mut self, id: &str) -> bool {
        let idx = self.items.iter().position(|i| i.id == id);
        if let Some(i) = idx {
            // Remove from outfits
            for outfit in &mut self.outfits {
                outfit.items.retain(|item_id| item_id != id);
            }
            self.items.remove(i);
            self.save();
            true
        } else {
            false
        }
    }

    fn get_item(&self, id: &str) -> Option<&Item> {
        self.items.iter().find(|i| i.id == id)
    }

    fn add_outfit(&mut self, name: String, item_ids: Vec<String>) -> Outfit {
        let outfit = Outfit {
            id: Uuid::new_v4().to_string(),
            name,
            items: item_ids,
            created: Utc::now().to_rfc3339(),
        };
        self.outfits.push(outfit.clone());
        self.save();
        outfit
    }

    fn search_items(&self, query: &str, category: &str, color: &str, season: &str) -> Vec<Item> {
        let mut results = self.items.clone();
        if !query.is_empty() {
            let q = query.to_lowercase();
            results.retain(|i| i.name.to_lowercase().contains(&q) || i.category.to_lowercase().contains(&q));
        }
        if !category.is_empty() {
            let c = category.to_lowercase();
            results.retain(|i| i.category.to_lowercase() == c);
        }
        if !color.is_empty() {
            let c = color.to_lowercase();
            results.retain(|i| i.color.to_lowercase() == c);
        }
        if !season.is_empty() {
            let s = season.to_lowercase();
            results.retain(|i| i.season.to_lowercase() == s || i.season.to_lowercase() == "all");
        }
        results
    }

    fn get_color_stats(&self) -> HashMap<String, usize> {
        let mut stats = HashMap::new();
        for item in &self.items {
            *stats.entry(item.color.clone()).or_insert(0) += 1;
        }
        stats
    }

    fn get_category_stats(&self) -> HashMap<String, usize> {
        let mut stats = HashMap::new();
        for item in &self.items {
            *stats.entry(item.category.clone()).or_insert(0) += 1;
        }
        stats
    }

    fn get_season_stats(&self) -> HashMap<String, usize> {
        let mut stats = HashMap::new();
        for item in &self.items {
            *stats.entry(item.season.clone()).or_insert(0) += 1;
        }
        stats
    }

    fn random_outfit(&self, season: &str) -> Vec<Item> {
        let mut items = self.items.clone();
        if !season.is_empty() && season.to_lowercase() != "all" {
            let s = season.to_lowercase();
            items.retain(|i| i.season.to_lowercase() == s || i.season.to_lowercase() == "all");
        }
        if items.is_empty() {
            return Vec::new();
        }
        let mut outfit = Vec::new();
        let mut rng = thread_rng();
        for cat in CATEGORIES {
            let cat_items: Vec<Item> = items.iter().filter(|i| i.category == *cat).cloned().collect();
            if !cat_items.is_empty() {
                if let Some(item) = cat_items.choose(&mut rng) {
                    outfit.push(item.clone());
                }
            }
        }
        outfit
    }
}

// ─── Main App ──────────────────────────────────────────────────────────────

struct WardrobeApp {
    data: WardrobeData,
}

impl WardrobeApp {
    fn new() -> Self {
        WardrobeApp { data: WardrobeData::new() }
    }

    fn ask(&self, prompt: &str) -> String {
        print!("{}", prompt);
        io::stdout().flush().unwrap();
        let mut line = String::new();
        io::stdin().read_line(&mut line).unwrap();
        line.trim().to_string()
    }

    fn show_menu(&self) {
        let total = self.data.items.len();
        let categories = self.data.get_category_stats().len();
        println!("\n{}", "═".repeat(50).cyan());
        println!("{}", c("👗 WARDROBE ORGANIZER", "bright cyan"));
        println!("{}", "═".repeat(50).cyan());
        println!("  Items: {}", total);
        println!("  Categories: {}", categories);
        println!("  Outfits: {}", self.data.outfits.len());
        println!("{}", "═".repeat(50).cyan());
        println!("  1. 👕 Add clothing item");
        println!("  2. 📋 List all items");
        println!("  3. 🔍 Search items");
        println!("  4. 📊 Statistics");
        println!("  5. 🎭 Outfit Builder");
        println!("  6. 🎲 Random Outfit");
        println!("  7. ❤️  Favorites (coming soon)");
        println!("  8. 🗑️  Delete item");
        println!("  0. 🚪 Exit");
        println!("{}", "═".repeat(50).cyan());
    }

    fn list_items(&self, items: Option<&[Item]>) {
        let target = match items {
            Some(i) => i,
            None => &self.data.items,
        };
        if target.is_empty() {
            println!("{}", c("No items found.", "yellow"));
            return;
        }
        println!("\n👕 WARDROBE ITEMS");
        println!("{}", "─".repeat(60).dimmed());
        for (i, item) in target.iter().enumerate() {
            let emoji = COLOR_EMOJIS.get(&item.color).unwrap_or(&"".to_string());
            println!("  {}. {} {} ({}) {} {}", i+1, emoji, item.name, item.category, item.color, item.season);
        }
    }

    fn add_item(&mut self) {
        let name = self.ask("Item name: ");
        println!("Categories: {}", CATEGORIES.join(", "));
        let category = self.ask("Category: ");
        println!("Colors: {}", COLORS.join(", "));
        let color = self.ask("Color: ");
        println!("Seasons: {}", SEASONS.join(", "));
        let season = self.ask("Season: ");
        let photo = self.ask("Photo path (optional): ");
        let item = self.data.add_item(name, category, color, season, photo);
        let emoji = COLOR_EMOJIS.get(&item.color).unwrap_or(&"".to_string());
        println!("{}", c(&format!("✅ Added {} {} ({}, {})", emoji, item.name, item.category, item.color), "green"));
    }

    fn search_items(&self) {
        let query = self.ask("Search term (name/category): ");
        let category = self.ask("Filter by category (optional): ");
        let color = self.ask("Filter by color (optional): ");
        let season = self.ask("Filter by season (optional): ");
        let results = self.data.search_items(&query, &category, &color, &season);
        if results.is_empty() {
            println!("{}", c("No items match your search.", "yellow"));
        } else {
            self.list_items(Some(&results));
        }
    }

    fn show_stats(&self) {
        let color_stats = self.data.get_color_stats();
        let category_stats = self.data.get_category_stats();
        let season_stats = self.data.get_season_stats();
        println!("\n📊 STATISTICS");
        println!("{}", "─".repeat(30).dimmed());
        println!("\n🎨 Colors:");
        for (color, count) in color_stats {
            let emoji = COLOR_EMOJIS.get(&color).unwrap_or(&"".to_string());
            println!("  {} {}: {}", emoji, color, count);
        }
        println!("\n📂 Categories:");
        for (cat, count) in category_stats {
            println!("  {}: {}", cat, count);
        }
        println!("\n🌦️ Seasons:");
        for (season, count) in season_stats {
            println!("  {}: {}", season, count);
        }
    }

    fn outfit_builder(&mut self) {
        if self.data.items.is_empty() {
            println!("{}", c("No items. Add some clothing first!", "yellow"));
            return;
        }
        let name = self.ask("Outfit name: ");
        println!("Select items for your outfit:");
        self.list_items(None);
        let mut item_ids = Vec::new();
        loop {
            let choice = self.ask("Enter item number to add (or 'done'): ");
            if choice.to_lowercase() == "done" { break; }
            if let Ok(idx) = choice.parse::<usize>() {
                let idx = idx - 1;
                if idx < self.data.items.len() {
                    let item = &self.data.items[idx];
                    if !item_ids.contains(&item.id) {
                        item_ids.push(item.id.clone());
                        println!("{}", c(&format!("Added {}", item.name), "green"));
                    } else {
                        println!("{}", c("Already added", "yellow"));
                    }
                } else {
                    println!("{}", c("Invalid number", "red"));
                }
            } else {
                println!("{}", c("Invalid input", "red"));
            }
        }
        if !item_ids.is_empty() {
            let outfit = self.data.add_outfit(name, item_ids);
            println!("{}", c(&format!("✅ Outfit '{}' created with {} items", outfit.name, outfit.items.len()), "green"));
            self.show_outfit(&outfit);
        }
    }

    fn show_outfit(&self, outfit: &Outfit) {
        println!("\n🎭 {}", outfit.name);
        for id in &outfit.items {
            if let Some(item) = self.data.get_item(id) {
                let emoji = COLOR_EMOJIS.get(&item.color).unwrap_or(&"".to_string());
                println!("  {} {} ({}) {}", emoji, item.name, item.category, item.color);
            }
        }
    }

    fn random_outfit(&self) {
        if self.data.items.is_empty() {
            println!("{}", c("No items. Add some clothing first!", "yellow"));
            return;
        }
        let season = self.ask("Season (optional, press Enter for all): ");
        let outfit = self.data.random_outfit(&season);
        if outfit.is_empty() {
            println!("{}", c("No items for this season.", "yellow"));
            return;
        }
        println!("\n🎲 Random Outfit");
        for item in outfit {
            let emoji = COLOR_EMOJIS.get(&item.color).unwrap_or(&"".to_string());
            println!("  {} {} ({}) {}", emoji, item.name, item.category, item.color);
        }
    }

    fn delete_item(&mut self) {
        if self.data.items.is_empty() {
            println!("{}", c("No items to delete.", "yellow"));
            return;
        }
        self.list_items(None);
        let choice = self.ask("Enter item number to delete (or 'cancel'): ");
        if choice.to_lowercase() == "cancel" { return; }
        if let Ok(idx) = choice.parse::<usize>() {
            let idx = idx - 1;
            if idx < self.data.items.len() {
                let item = &self.data.items[idx];
                let confirm = self.ask(&format!("Delete '{}'? (yes/no): ", item.name));
                if confirm.to_lowercase() == "yes" {
                    self.data.delete_item(&item.id);
                    println!("{}", c(&format!("🗑️  Deleted {}", item.name), "yellow"));
                }
            } else {
                println!("{}", c("Invalid number", "red"));
            }
        } else {
            println!("{}", c("Invalid input", "red"));
        }
    }

    fn run(&mut self) {
        println!("{}", "\n👗 Wardrobe Organizer – Smart Clothing Manager".bright().cyan());
        println!("{}", "Catalog your wardrobe, plan outfits!".dimmed());

        loop {
            self.show_menu();
            let choice = self.ask("Your choice: ");
            match choice.as_str() {
                "1" => self.add_item(),
                "2" => self.list_items(None),
                "3" => self.search_items(),
                "4" => self.show_stats(),
                "5" => self.outfit_builder(),
                "6" => self.random_outfit(),
                "7" => println!("{}", c("❤️ Favorites feature coming soon!", "dim")),
                "8" => self.delete_item(),
                "0" => {
                    println!("{}", c("👋 Stay stylish! Goodbye!", "cyan"));
                    return;
                }
                _ => println!("{}", c("❌ Invalid choice.", "red")),
            }
            if choice != "0" {
                print!("\nPress Enter to continue...");
                io::stdout().flush().unwrap();
                let mut _dummy = String::new();
                io::stdin().read_line(&mut _dummy).unwrap();
            }
        }
    }
}

// ─── Main ────────────────────────────────────────────────────────────────────

#[macro_use] extern crate lazy_static;

fn main() {
    let mut app = WardrobeApp::new();
    app.run();
}
